src/rules/store/no-store-subscription.ts

```diff
@@ -31,9 +31,7 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
     if (!storeName) return {}
 
     return {
-      [`CallExpression:has(${storeExpression(
-        storeName,
-      )}) > MemberExpression > Identifier[name='subscribe']`](
+      [`CallExpression:matches([callee.object.callee.object.name=${storeName}], [callee.object.callee.object.object.type='ThisExpression'][callee.object.callee.object.property.name=${storeName}]) > MemberExpression > Identifier[name='subscribe']`](
```

Comment:
Unfortunately I don't think we can reuse the `storeExpression` function anymore, but let me know if you have any suggestions.

Reply:
I think you're right.
Does this mean that we can remove `storeExpression` ?